components/camel-jetty9/src/main/java/org/apache/camel/component/jetty9/JettyContentExchange9.java

```diff
@@ -214,14 +220,28 @@ public void onFailure(Request request, Throwable failure) {
             }
 
         };
-        BufferingResponseListener responseListener = new BufferingResponseListener() {
+
+        InputStreamResponseListener responseListener = new InputStreamResponseListener() {
+            ByteArrayOutputStream baos = new ByteArrayOutputStream();
+
+            @Override
+            public void onContent(Response response, ByteBuffer content, Callback callback) {
+                if (response.getStatus() != SC_OK) {
```

Comment:
Why this WARN log, wont you be able to get some error and still get data. I would remove that WARN